Sources/SwiftSyntax/SourceEdit.swift

```diff
@@ -13,24 +13,68 @@
 /// A textual edit to the original source represented by a range and a
 /// replacement.
 public struct SourceEdit: Equatable, Sendable {
-  /// The half-open range that this edit applies to.
-  public let range: Range<AbsolutePosition>
-  /// The text to replace the original range with. Empty for a deletion.
-  public let replacement: String
-
-  /// Length of the original source range that this edit applies to. Zero if
-  /// this is an addition.
-  public var length: SourceLength {
-    return SourceLength(utf8Length: range.upperBound.utf8Offset - range.lowerBound.utf8Offset)
+  /// The byte range of the original source buffer that the edit applies to.
+  public let range: ByteSourceRange
+
+  /// The UTF-8 bytes that should be inserted as part of the edit
+  public let replacement: [UInt8]
+
+  /// A string representation of the replacement
+  public var replacementText: String { return String(decoding: replacement, as: UTF8.self) }
```

Comment:
To not make this a source-breaking change, this would need to continue being called `replacement`. I’d call the `[UInt8]` version `replacementBytes`.

We don’t need to be API-compatible with `IncrementalEdit.replacement` because that property only gets introduced in swift-syntax 600.